**PRIMER PARCIAL IA EN GOOGLE COLAB**

Magaly Escalera Alba

In [17]:
!pip install spacy
!python -m spacy download es_core_news_sm
import nltk
nltk.download('punkt')
nltk.download('punkt_tab')

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.9/12.9 MB 46.0 MB/s eta 0:00:00
✔ Download and installation successful
You can now load the package via spacy.load('es_core_news_sm')
⚠ Restart to reload dependencies
If you are in a Jupyter or Colab notebook, you may need to restart Python in
order to load all the package's dependencies. You can do this by selecting the
'Restart kernel' or 'Restart runtime' option.


[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package punkt_tab to /root/nltk_data...
[nltk_data]   Package punkt_tab is already up-to-date!


True

In [18]:
import nltk
import spacy

# Cargar modelo de español
nlp = spacy.load("es_core_news_sm")

def analisis_morfologico(oracion):
    """
    Realiza análisis morfológico completo de una oración en español
    con salida similar al ejemplo del examen.
    """

    # 1️⃣ Tokenizar con NLTK
    tokens = nltk.word_tokenize(oracion, language="spanish")

    # 2️⃣ Conteos
    total_tokens = len(tokens)
    total_tipos = len(set(tokens))
    ratio_tt = round(total_tipos / total_tokens, 3)

    # 3️⃣ Etiquetado morfosintáctico con spaCy
    doc = nlp(oracion)
    pos_tags = [(token.text, token.pos_) for token in doc]

    # 4️⃣ Estructura final
    return {
        "total_tokens": total_tokens,
        "total_tipos": total_tipos,
        "ratio_tt": ratio_tt,
        "pos_tags": pos_tags
    }

In [19]:
oracion_ejemplo = "El gato negro salta alto y el perro corre rápido por el parque."
resultado = analisis_morfologico(oracion_ejemplo)

import pprint
pprint.pprint(resultado)



{'pos_tags': [('El', 'DET'),
              ('gato', 'NOUN'),
              ('negro', 'ADJ'),
              ('salta', 'NOUN'),
              ('alto', 'ADJ'),
              ('y', 'CCONJ'),
              ('el', 'DET'),
              ('perro', 'PROPN'),
              ('corre', 'VERB'),
              ('rápido', 'ADV'),
              ('por', 'ADP'),
              ('el', 'DET'),
              ('parque', 'NOUN'),
              ('.', 'PUNCT')],
 'ratio_tt': 0.929,
 'total_tipos': 13,
 'total_tokens': 14}


Análisis de resultados:
El análisis morfológico de la oración muestra que contiene 14 tokens y 13 tipos únicos, lo que da un ratio tipo-token de 0.929. Esto indica que la oración tiene una variedad alta de palabras únicas respecto al total de tokens, lo que refleja diversidad léxica.

Las etiquetas POS asignadas permiten identificar correctamente las funciones gramaticales de cada palabra: por ejemplo, sustantivos (NOUN, PROPN), verbos (VERB), adjetivos (ADJ), adverbios (ADV), determinantes (DET) y conjunciones (CCONJ). Esta información es útil para análisis sintáctico, extracción de información y procesamiento de lenguaje natural en español, ya que permite diferenciar el rol de cada palabra dentro de la oración.